internal/impl/mail/input_pop3.go

```diff
@@ -0,0 +1,299 @@
+package mail
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/benthosdev/benthos/v4/public/service"
+	"github.com/knadh/go-pop3"
+)
+
+const (
+	hostParam              = "host"
+	portParam              = "port"
+	usernameParam          = "username"
+	passwordParam          = "password"
+	tlsParam               = "tls"
+	tlsEnabledParam        = "enabled"
+	tlsSkipCertVerifyParam = "skip_cert_verify"
+	pullParam              = "pull"
+	countParam             = "count"
+	intervalParam          = "interval"
+	maxInFlightParam       = "max_in_flight"
+	timeoutParam           = "timeout"
+	maxBodySizeParam       = "max_body_size"
+	listPageSizeParam      = "list_page_size"
+)
+
+func pop3ConfigSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Categories("Services").
+		Version("4.12.0").
+		Summary("Connects to a POP3 server and periodically reads the emails.").
+		Description(`
+Reads emails from a POP3 server. The emails are permanently deleted when delivered.
+
+### Metadata
+
+This input adds the following metadata fields to each message if present:
+
+` + "```" + `
+- Subject
+- From
+- To
+- Cc
+- Bcc
+- Date
+- Received
+- Content-Type
+- Return-Path
+- Delivered-To
+` + "```" + `
+You can access these metadata fields using [function interpolation](/docs/configuration/interpolation#bloblang-queries).
+`).
+		Field(service.NewStringField(hostParam).
+			Description("The host of the POP3 server").
+			Example("pop.gmail.com")).
+		Field(service.NewIntField(portParam).
+			Description("The port of the POP3 server.").
+			Default(995)).
+		Field(service.NewStringField(usernameParam).
+			Description("The username to login on the POP3 server.")).
+		Field(service.NewStringField(passwordParam).
+			Description("The password to login on the POP3 server.")).
+		Field(service.NewObjectField(tlsParam,
+			service.NewBoolField(tlsEnabledParam).
+				Description(`Whether TLS is enabled.`),
+			service.NewBoolField(tlsSkipCertVerifyParam).
+				Description(`Whether to skip server side certificate verification.`)).Advanced()).
+		Field(service.NewObjectField(pullParam,
+			service.NewIntField(countParam).
+				Description(`An optional number of pulls to fetch messages, if set above 0 the specified number of pulls is executed and then the input will shut down.`).
```

Comment:
This is an interesting one, we could achieve the same thing by placing this input within a `read_until` input, not sure it's worth having an explicit field here.